<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
    <title>lemon 0717测试报告</title>
    <meta name="generator" content="HTMLTestRunner 0.8.2.2"/>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
    <link href="http://libs.baidu.com/bootstrap/3.0.3/css/bootstrap.min.css" rel="stylesheet">
    <script src="http://libs.baidu.com/jquery/2.0.0/jquery.min.js"></script>
    <script src="http://libs.baidu.com/bootstrap/3.0.3/js/bootstrap.min.js"></script>
    
<style type="text/css" media="screen">
body        { font-family: Microsoft YaHei,Tahoma,arial,helvetica,sans-serif;padding: 20px; font-size: 120%; }
table       { font-size: 100%; }
/* -- heading ---------------------------------------------------------------------- */
.heading {
    margin-top: 0ex;
    margin-bottom: 1ex;
}
.heading .description {
    margin-top: 4ex;
    margin-bottom: 6ex;
}
/* -- report ------------------------------------------------------------------------ */
#total_row  { font-weight: bold; }
.passCase   { color: #5cb85c; }
.failCase   { color: #d9534f; font-weight: bold; }
.errorCase  { color: #f0ad4e; font-weight: bold; }
.hiddenRow  { display: none; }
.testcase   { margin-left: 2em; }
</style>

</head>
<body >
<script language="javascript" type="text/javascript">
output_list = Array();
/*level 调整增加只显示通过用例的分类 --Findyou
0:Summary //all hiddenRow
1:Failed  //pt hiddenRow, ft none
2:Pass    //pt none, ft hiddenRow
3:All     //pt none, ft none
*/
function showCase(level) {
    trs = document.getElementsByTagName("tr");
    for (var i = 0; i < trs.length; i++) {
        tr = trs[i];
        id = tr.id;
        if (id.substr(0,2) == 'ft') {
            if (level == 2 || level == 0 ) {
                tr.className = 'hiddenRow';
            }
            else {
                tr.className = '';
            }
        }
        if (id.substr(0,2) == 'pt') {
            if (level < 2) {
                tr.className = 'hiddenRow';
            }
            else {
                tr.className = '';
            }
        }
    }
    //加入【详细】切换文字变化 --Findyou
    detail_class=document.getElementsByClassName('detail');
	//console.log(detail_class.length)
	if (level == 3) {
		for (var i = 0; i < detail_class.length; i++){
			detail_class[i].innerHTML="收起"
		}
	}
	else{
			for (var i = 0; i < detail_class.length; i++){
			detail_class[i].innerHTML="详细"
		}
	}
}
function showClassDetail(cid, count) {
    var id_list = Array(count);
    var toHide = 1;
    for (var i = 0; i < count; i++) {
        //ID修改 点 为 下划线 -Findyou
        tid0 = 't' + cid.substr(1) + '_' + (i+1);
        tid = 'f' + tid0;
        tr = document.getElementById(tid);
        if (!tr) {
            tid = 'p' + tid0;
            tr = document.getElementById(tid);
        }
        id_list[i] = tid;
        if (tr.className) {
            toHide = 0;
        }
    }
    for (var i = 0; i < count; i++) {
        tid = id_list[i];
        //修改点击无法收起的BUG，加入【详细】切换文字变化 --Findyou
        if (toHide) {
            document.getElementById(tid).className = 'hiddenRow';
            document.getElementById(cid).innerText = "详细"
        }
        else {
            document.getElementById(tid).className = '';
            document.getElementById(cid).innerText = "收起"
        }
    }
}
function html_escape(s) {
    s = s.replace(/&/g,'&amp;');
    s = s.replace(/</g,'&lt;');
    s = s.replace(/>/g,'&gt;');
    return s;
}
</script>
<div class='heading'>
<h1 style="font-family: Microsoft YaHei">lemon 0717测试报告</h1>
<p class='attribute'><strong>测试人员 : </strong> 最爱宝宝的Tuosang</p>
<p class='attribute'><strong>开始时间 : </strong> 2019-07-24 17:21:51</p>
<p class='attribute'><strong>合计耗时 : </strong> 0:00:00.581041</p>
<p class='attribute'><strong>测试结果 : </strong> 共 5，错误 5，通过率= 0.00%</p>

<p class='description'>lemon 0717测试报告</p>
</div>


<p id='show_detail_line'>
<a class="btn btn-primary" href='javascript:showCase(0)'>概要{ 0.00% }</a>
<a class="btn btn-danger" href='javascript:showCase(1)'>失败{ 0 }</a>
<a class="btn btn-success" href='javascript:showCase(2)'>通过{ 0 }</a>
<a class="btn btn-info" href='javascript:showCase(3)'>所有{ 5 }</a>
</p>
<table id='result_table' class="table table-condensed table-bordered table-hover">
<colgroup>
<col align='left' />
<col align='right' />
<col align='right' />
<col align='right' />
<col align='right' />
<col align='right' />
</colgroup>
<tr id='header_row' class="text-center success" style="font-weight: bold;font-size: 16px;">
    <td>用例集/测试用例</td>
    <td>总计</td>
    <td>通过</td>
    <td>失败</td>
    <td>错误</td>
    <td>详细</td>
</tr>

<tr class='errorClass warning'>
    <td>API_6.test_cases.test_register.TestCases</td>
    <td class="text-center">2</td>
    <td class="text-center">0</td>
    <td class="text-center">0</td>
    <td class="text-center">2</td>
    <td class="text-center"><a href="javascript:showClassDetail('c1',2)" class="detail" id='c1'>详细</a></td>
</tr>

<tr id='ft1_1' class='none'>
    <td class='errorCase'><div class='testcase'>test_cases_1</div></td>
    <td colspan='5' align='center'>
    <!--默认收起错误信息 -Findyou
    <button id='btn_ft1_1' type="button"  class="btn btn-danger btn-xs collapsed" data-toggle="collapse" data-target='#div_ft1_1'>错误</button>
    <div id='div_ft1_1' class="collapse">  -->
    <!-- 默认展开错误信息 -Findyou -->
    <button id='btn_ft1_1' type="button"  class="btn btn-danger btn-xs" data-toggle="collapse" data-target='#div_ft1_1'>错误</button>
    <div id='div_ft1_1' class="collapse in" align="left">
    <pre>
    
ft1_1: 2019-07-24 17:21:51,853-INFO-my_log.py-api-log-[日志信息]:------正在测试Register模里面第1条测试用例：正常注册
2019-07-24 17:21:51,854-INFO-my_log.py-api-log-[日志信息]:测试数据是：{'CaseId': 1, 'Method': 'get', 'Params': "{'mobilephone':'15096098898','pwd':'123456','regname':'lemonchupan'}", 'Title': '正常注册', 'ExpectedResult': "{'msg': '手机号码已被注册', 'status': 0, 'data': None, 'code': '20110'}", 'Url': 'http://47.107.168.87:8080/futureloan/mvc/api/member/register', 'sql': "{'msg': '手机号码已被注册', 'status': 0, 'data': None, 'code': '20110'}", 'Module': 'Register'}
Traceback (most recent call last):
  File "C:\Python34\lib\site-packages\ddt.py", line 145, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\kilimall\PycharmProjects\untitled1\python lesson1\API_6\test_cases\test_register.py", line 35, in test_cases
    resp = HttpRequest().http_request(method, url, param)
TypeError: http_request() missing 1 required positional argument: 'cookies'


    </pre>
    </div>
    </td>
</tr>

<tr id='ft1_2' class='none'>
    <td class='errorCase'><div class='testcase'>test_cases_2</div></td>
    <td colspan='5' align='center'>
    <!--默认收起错误信息 -Findyou
    <button id='btn_ft1_2' type="button"  class="btn btn-danger btn-xs collapsed" data-toggle="collapse" data-target='#div_ft1_2'>错误</button>
    <div id='div_ft1_2' class="collapse">  -->
    <!-- 默认展开错误信息 -Findyou -->
    <button id='btn_ft1_2' type="button"  class="btn btn-danger btn-xs" data-toggle="collapse" data-target='#div_ft1_2'>错误</button>
    <div id='div_ft1_2' class="collapse in" align="left">
    <pre>
    
ft1_2: 2019-07-24 17:21:51,856-INFO-my_log.py-api-log-[日志信息]:------正在测试Register模里面第2条测试用例：手机号为空
2019-07-24 17:21:51,857-INFO-my_log.py-api-log-[日志信息]:测试数据是：{'CaseId': 2, 'Method': 'get', 'Params': "{'mobilephone':'','pwd':'123456','regname':'lemonchupan'}", 'Title': '手机号为空', 'ExpectedResult': "{'data': None, 'status': 0, 'code': '20103', 'msg': '手机号不能为空'}", 'Url': 'http://47.107.168.87:8080/futureloan/mvc/api/member/register', 'Module': 'Register'}
Traceback (most recent call last):
  File "C:\Python34\lib\site-packages\ddt.py", line 145, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\kilimall\PycharmProjects\untitled1\python lesson1\API_6\test_cases\test_register.py", line 35, in test_cases
    resp = HttpRequest().http_request(method, url, param)
TypeError: http_request() missing 1 required positional argument: 'cookies'


    </pre>
    </div>
    </td>
</tr>

<tr class='errorClass warning'>
    <td>API_6.test_cases.test_recharge.TestCases</td>
    <td class="text-center">3</td>
    <td class="text-center">0</td>
    <td class="text-center">0</td>
    <td class="text-center">3</td>
    <td class="text-center"><a href="javascript:showClassDetail('c2',3)" class="detail" id='c2'>详细</a></td>
</tr>

<tr id='ft2_1' class='none'>
    <td class='errorCase'><div class='testcase'>test_cases_1</div></td>
    <td colspan='5' align='center'>
    <!--默认收起错误信息 -Findyou
    <button id='btn_ft2_1' type="button"  class="btn btn-danger btn-xs collapsed" data-toggle="collapse" data-target='#div_ft2_1'>错误</button>
    <div id='div_ft2_1' class="collapse">  -->
    <!-- 默认展开错误信息 -Findyou -->
    <button id='btn_ft2_1' type="button"  class="btn btn-danger btn-xs" data-toggle="collapse" data-target='#div_ft2_1'>错误</button>
    <div id='div_ft2_1' class="collapse in" align="left">
    <pre>
    
ft2_1: 2019-07-24 17:21:51,859-INFO-my_log.py-api-log-[日志信息]:------正在测试login模里面第1条测试用例：正常登陆
2019-07-24 17:21:51,860-INFO-my_log.py-api-log-[日志信息]:测试数据是：{'CaseId': 1, 'Method': 'get', 'Params': "{'mobilephone':'18574379913','pwd':'123456'}", 'Title': '正常登陆', 'ExpectedResult': "{'status':1,'code':'10001','data': None,'msg':'登录成功'}", 'Url': 'http://47.107.168.87:8080/futureloan/mvc/api/member/login', 'Module': 'login'}
Traceback (most recent call last):
  File "C:\Users\kilimall\PycharmProjects\untitled1\python lesson1\API_6\test_cases\test_recharge.py", line 44, in test_cases
    self.assertEqual(eval(case['ExpectedResult']),resp.json())#http发送请求拿到的实际返回结果
  File "C:\Python34\lib\site-packages\requests\models.py", line 897, in json
    return complexjson.loads(self.text, **kwargs)
  File "C:\Python34\lib\json\__init__.py", line 318, in loads
    return _default_decoder.decode(s)
  File "C:\Python34\lib\json\decoder.py", line 343, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Python34\lib\json\decoder.py", line 361, in raw_decode
    raise ValueError(errmsg("Expecting value", s, err.value)) from None
ValueError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python34\lib\site-packages\ddt.py", line 145, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\kilimall\PycharmProjects\untitled1\python lesson1\API_6\test_cases\test_recharge.py", line 53, in test_cases
    self.t.write_back(case['CaseId'] + 1, 9, TestResult)
NameError: name 'TestResult' is not defined


    </pre>
    </div>
    </td>
</tr>

<tr id='ft2_2' class='none'>
    <td class='errorCase'><div class='testcase'>test_cases_2</div></td>
    <td colspan='5' align='center'>
    <!--默认收起错误信息 -Findyou
    <button id='btn_ft2_2' type="button"  class="btn btn-danger btn-xs collapsed" data-toggle="collapse" data-target='#div_ft2_2'>错误</button>
    <div id='div_ft2_2' class="collapse">  -->
    <!-- 默认展开错误信息 -Findyou -->
    <button id='btn_ft2_2' type="button"  class="btn btn-danger btn-xs" data-toggle="collapse" data-target='#div_ft2_2'>错误</button>
    <div id='div_ft2_2' class="collapse in" align="left">
    <pre>
    
ft2_2: 2019-07-24 17:21:52,062-INFO-my_log.py-api-log-[日志信息]:------正在测试recharge模里面第2条测试用例：正常充值
2019-07-24 17:21:52,064-INFO-my_log.py-api-log-[日志信息]:测试数据是：{'CaseId': 2, 'Method': 'post', 'Params': "{'mobilephone':'18574379913','amount':'10000'}", 'Title': '正常充值', 'ExpectedResult': '{"status":1,"code":"10001","data":{"id":84,"regname":"lemoncp","pwd":"E10ADC3949BA59ABBE56E057F20F883E","mobilephone":"18574379913","leaveamount":"90000.00","type":"1","regtime":"2019-07-15 15:05:07.0"},"msg":"充值成功"}', 'Url': 'http://47.107.168.87:8080/futureloan/mvc/api/member/recharge', 'Module': 'recharge'}
Traceback (most recent call last):
  File "C:\Users\kilimall\PycharmProjects\untitled1\python lesson1\API_6\test_cases\test_recharge.py", line 44, in test_cases
    self.assertEqual(eval(case['ExpectedResult']),resp.json())#http发送请求拿到的实际返回结果
  File "C:\Python34\lib\site-packages\requests\models.py", line 897, in json
    return complexjson.loads(self.text, **kwargs)
  File "C:\Python34\lib\json\__init__.py", line 318, in loads
    return _default_decoder.decode(s)
  File "C:\Python34\lib\json\decoder.py", line 343, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Python34\lib\json\decoder.py", line 361, in raw_decode
    raise ValueError(errmsg("Expecting value", s, err.value)) from None
ValueError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python34\lib\site-packages\ddt.py", line 145, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\kilimall\PycharmProjects\untitled1\python lesson1\API_6\test_cases\test_recharge.py", line 53, in test_cases
    self.t.write_back(case['CaseId'] + 1, 9, TestResult)
NameError: name 'TestResult' is not defined


    </pre>
    </div>
    </td>
</tr>

<tr id='ft2_3' class='none'>
    <td class='errorCase'><div class='testcase'>test_cases_3</div></td>
    <td colspan='5' align='center'>
    <!--默认收起错误信息 -Findyou
    <button id='btn_ft2_3' type="button"  class="btn btn-danger btn-xs collapsed" data-toggle="collapse" data-target='#div_ft2_3'>错误</button>
    <div id='div_ft2_3' class="collapse">  -->
    <!-- 默认展开错误信息 -Findyou -->
    <button id='btn_ft2_3' type="button"  class="btn btn-danger btn-xs" data-toggle="collapse" data-target='#div_ft2_3'>错误</button>
    <div id='div_ft2_3' class="collapse in" align="left">
    <pre>
    
ft2_3: 2019-07-24 17:21:52,237-INFO-my_log.py-api-log-[日志信息]:------正在测试recharge模里面第3条测试用例：不输入充值金额
2019-07-24 17:21:52,239-INFO-my_log.py-api-log-[日志信息]:测试数据是：{'CaseId': 3, 'Method': 'post', 'Params': "{'mobilephone':'18574379913','amount':''}", 'Title': '不输入充值金额', 'ExpectedResult': '{"status":0,"code":"20115","data":None,"msg":"请输入金额"}', 'Url': 'http://47.107.168.87:8080/futureloan/mvc/api/member/recharge', 'Module': 'recharge'}
Traceback (most recent call last):
  File "C:\Users\kilimall\PycharmProjects\untitled1\python lesson1\API_6\test_cases\test_recharge.py", line 44, in test_cases
    self.assertEqual(eval(case['ExpectedResult']),resp.json())#http发送请求拿到的实际返回结果
  File "C:\Python34\lib\site-packages\requests\models.py", line 897, in json
    return complexjson.loads(self.text, **kwargs)
  File "C:\Python34\lib\json\__init__.py", line 318, in loads
    return _default_decoder.decode(s)
  File "C:\Python34\lib\json\decoder.py", line 343, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Python34\lib\json\decoder.py", line 361, in raw_decode
    raise ValueError(errmsg("Expecting value", s, err.value)) from None
ValueError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python34\lib\site-packages\ddt.py", line 145, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\kilimall\PycharmProjects\untitled1\python lesson1\API_6\test_cases\test_recharge.py", line 53, in test_cases
    self.t.write_back(case['CaseId'] + 1, 9, TestResult)
NameError: name 'TestResult' is not defined


    </pre>
    </div>
    </td>
</tr>

<tr id='total_row' class="text-center active">
    <td>总计</td>
    <td>5</td>
    <td>0</td>
    <td>0</td>
    <td>5</td>
    <td>通过率：0.00%</td>
</tr>
</table>

<div id='ending'>&nbsp;</div>
    <div style=" position:fixed;right:50px; bottom:30px; width:20px; height:20px;cursor:pointer">
    <a href="#"><span class="glyphicon glyphicon-eject" style = "font-size:30px;" aria-hidden="true">
    </span></a></div>
    
</body>
</html>
